### DISCUSSION TITLE
[Idea] ChatGPT conversation as a graph

### DISCUSSION THREAD

**Discussion by zikajk** [karthink/gptel#117] [Ideas]
Hello,

I frequently engage in extended discussions with ChatGPT, often diving deep into specific details. However, I've found that these in-depth conversations can become lengthy and hard to navigate. And creating a new conversation thread will lose context so I don't do that. 

Here's an idea I've been considering: Would it be possible to use Emacs, in conjunction with your package, to create a more organized interaction with ChatGPT? My thought is to have a main conversation buffer and then branch out to topic-specific buffers when seeking detailed information. This would allow for a modular discussion, making it easier to navigate between the main topic and its subtopics.

For instance, if I ask ChatGPT about the "10 largest lakes in Europe", it would provide a list. I could then mark "Lake Ladoga", use a function like `(give-me-details)`, and a new buffer focused on "Lake Ladoga" would open where I can continue with the conversation on this topic. Within this buffer, the conversation about this specific lake would ensue. Ideally, there should be some built-in logic: upon using main buffer again, a prompt like "thanks for the details, let's return to the discussion on the 10 largest lakes" would be sent to ChatGPT, and vice versa. 
_Visually, this could resemble a ROAM-style map, with "10-largest-lake" as the main node and "Lake Ladoga Topic" as a connected sub-node._

Do you think this approach is feasible with the gptel package? I believe it could enhance the user experience when having detailed conversations with ChatGPT.

Edit: `(give-me-details)` might offer a list of choices as customizable prompts.

**Comment by karthink** [comment]
You can already do some of this with `gptel-set-topic`, which limits context by Org heading.  You can write some glue code that creates an Org link from selected text to a new heading with that heading.

**Reply by zikajk** [reply]
I've finally found a moment to try this out. I didn't read documentation enough.

Basically, I don't have to add anything - it works perfectly. Sub-headings already behave like subtopics, and I can always go back up and inquire in the main topic (top heading). I had no idea it works this well. Thank you!

**Reply by zikajk** [reply]
https://www.loom.com/share/6505efd5238147e09b71aacbe5e62de5
